client/my-sites/media-library/filter-bar.jsx

```diff
@@ -1,110 +1,100 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	includes = require( 'lodash/includes' ),
-	noop = require( 'lodash/noop' );
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import {
+	includes,
+	noop,
+	map,
+} from 'lodash';
 
 /**
  * Internal dependencies
  */
-var SectionNav = require( 'components/section-nav' ),
-	SectionNavTabs = require( 'components/section-nav/tabs' ),
-	SectionNavTabItem = require( 'components/section-nav/item' ),
-	Search = require( 'components/search' );
+import SectionNav from 'components/section-nav';
+import SectionNavTabs from 'components/section-nav/tabs';
+import Search from 'components/search';
+import FilterItem from './filter-item';
 
-module.exports = React.createClass( {
-	displayName: 'MediaLibraryFilterBar',
-
-	propTypes: {
-		site: React.PropTypes.object,
+class MediaLibraryFilterBar extends Component {
+	static propTypes = {
 		basePath: React.PropTypes.string,
+		enabledFilters: React.PropTypes.arrayOf( React.PropTypes.string ),
 		filter: React.PropTypes.string,
 		filterRequiresUpgrade: React.PropTypes.bool,
-		enabledFilters: React.PropTypes.arrayOf( React.PropTypes.string ),
 		search: React.PropTypes.string,
+		site: React.PropTypes.object,
 		onFilterChange: React.PropTypes.func,
-		onSearch: React.PropTypes.func
-	},
-
-	getDefaultProps: function() {
-		return {
-			filter: '',
-			basePath: '/media',
-			onFilterChange: noop,
-			onSearch: noop
-		};
-	},
+		onSearch: React.PropTypes.func,
+	};
 
-	getSearchPlaceholderText: function() {
-		var placeholderText;
+	static defaultProps ={
+		filter: '',
+		basePath: '/media',
+		onFilterChange: noop,
+		onSearch: noop
+	};
 
-		switch ( this.props.filter ) {
+	getSearchPlaceholderText() {
+		const { filter, translate } = this.props;
+		switch ( filter ) {
 			case 'images':
-				placeholderText = this.translate( 'Search images…', { textOnly: true } );
-				break;
+				return translate( 'Search images…', { textOnly: true } );
```

Comment:
@yoavf If I'm not wrong I saw a comment of you talking about that textOnly isn't needed. is it correct?